pkg/controller/nodes/task/catalog/datacatalog/datacatalog.go

```diff
@@ -240,6 +242,21 @@ func (m *CatalogClient) Put(ctx context.Context, key catalog.Key, reader io.Outp
 		outputs = retOutputs
 	}
 
+	if reader.GetDeckPath() != nil {
+		lt := &core.Literal{
+			Value: &core.Literal_Scalar{
+				Scalar: &core.Scalar{
+					Value: &core.Scalar_Blob{Blob: &core.Blob{Uri: reader.GetDeckPath().String()}},
+				},
+			},
+		}
+		if outputs.Literals == nil {
+			outputs.Literals = map[string]*core.Literal{FlyteDeckFile: lt}
+		} else {
+			outputs.Literals[FlyteDeckFile] = lt
+		}
```

Comment:
I don't feel great about just throwing it as another output... 
How about we add another level here?
like, old:
```json
{
   "o1": ...,
   "o2": ...,
   "FLYTE_DECK_FILE": ...
}
```

I'm suggesting we make it:
```json
{
   "outputs": {
     "o1": ...,
     "o2": ...
   },
   "FLYTE_DECK_FILE": ...
}
```

wdyt @kumare3, @hamersaw

Reply:
Thinking about it again, DataCatalog already supports that leveling... 
Each ArtifactDataset can have multiple Artifacts attached to it.... today we create a DataSet per cacheKey (minus the inputs' hash) and then an Artifact attached to that per inputs' hash (i.e. a given task will have a single DataSet but for every invocation of that task, we will create a separate Artifact)
I think instead of adding the Deck URL as an additional output to the same artifact, we should, instead, add another Artifact to the same DataSet with a different tag (maybe `<inputs' hash>-flyte-deck` or something)
So you will basically have to call the `m.CreateArtifact()` a second time passing the `lt` variable you created here...
When we retrieve the artifacts [here](https://github.com/flyteorg/flytepropeller/blob/b9fb6e31eeb27a15c522995b0c0ea0f5536c0fe0/pkg/controller/nodes/task/catalog/datacatalog/datacatalog.go#L112-L122) we should then also try to retrieve the other artifact...

Pros:
* No naming collision, potential weird behavior when retrieving the artifacts

Cons:
* Additional API calls with potential of partial failures and added latency

Reply:
Ohh absolutely not an output. I actually want this to be a meta output, remember, I think a deck is possible even for partial executions

Reply:
@pingsutw, Maybe we drop it from catalog for now? this can happen later on when we figure out a good way of doing it?